src/conjuring/visibility.py

```diff
@@ -21,8 +23,13 @@ def is_git_repo() -> bool:
     return Path(".git").exists()
 
 
-def has_pyproject_toml() -> bool:
-    return Path("pyproject.toml").exists()
+def is_poetry_project() -> bool:
+    fpath = Path("pyproject.toml")
+    return fpath.exists() and _has_poetry_line(fpath)
+
+
+def _has_poetry_line(fpath: Path) -> bool:
+    return any(re.search(POETRY_LINE, line) for line in fpath.open())
```

Comment:
You can use `fpath.read_text()` to read the whole file at once.

Reply:
I admit this is likely unneeded optimization. Then again I didn't want to do string matching for whole file contents, sort of learned the idiom as the same one-liner scales to huge files (and quits early)

Reply:
> (and quits early)

Ah, yes, agreed, you're right. 👍🏻 
Let's keep it like this then.